### PR TITLE
GEODE-8870: Removes GFE_70.

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
The PR is intentionally broken into multiple commits. The work to just remove the version is in the first commit. 

The followup commits are to cleanup all the compiler and static analyzer warnings in the changed files.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/test/java/org/apache/geode/internal/DataSerializableJUnitTest.java
Comment (DonalEvans) [inline geode-core/src/test/java/org/apache/geode/internal/DataSerializableJUnitTest.java]: Is there a specific reason for this version to be used here? Similarly for the use of `GEODE_1_12_0` below.

File: geode-core/src/main/java/org/apache/geode/internal/cache/Oplog.java
Comment (DonalEvans) [reply geode-core/src/main/java/org/apache/geode/internal/cache/Oplog.java:5631]: Sounds good. I'll approve the PR now, on the assumption that the commit will get pushed before it's merged.

**Review by DonalEvans — APPROVED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/persistence/DiskInitFileParser.java
Comment (upthewaterspout) [inline geode-core/src/main/java/org/apache/geode/internal/cache/persistence/DiskInitFileParser.java:584]: In general, when we read from disk or the network we should be creating a VersionedDataInputStream. That allows us to make DataSerializableFixedId objects backwards compatible just by adding special fromData_Pre_XXX methods.

Maybe we should create one of those here, using gfVersion.

**Review by upthewaterspout — COMMENTED** [review]
Thanks for cleaning things up! I reviewed my codeowner files and had one comment, inline.

File: geode-core/src/main/java/org/apache/geode/internal/cache/persistence/DiskInitFileParser.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/persistence/DiskInitFileParser.java:584]: I think if this method needs versioning support added it should come when there is a version change here or a separate PR to add versioning support and tests. The only change here was to pull out the obsolete version check which left us with the original un-versioned from data call.

Going through all these cleanups and digging deep into areas of code I haven't been I have noticed there is inconsistent use of `VersionedDataInputStream` and `VersionedDataSerializable`. I wonder if we should have an effort to normalize on the versioned versions (🥁) and deprecate the older stuff.

**Review by echobravopapa — APPROVED** [review]
my codeowner files look good, thanks for the cleanup

**Comment by agingade** [comment]
I see lot more refactoring changes than the fix for actual issue. I agree and believe in incremental refactoring as we come across the opportunity; but here I see lot more refactoring changes; even in non-related files (where the actual change is not expected); it will be nice to keep those changes minimal and focused on the issue being fixed...just my thoughts and I am fine with other way too...

**Comment by pivotal-jbarrett** [comment]
> I see lot more refactoring changes than the fix for actual issue. I agree and believe in incremental refactoring as we come across the opportunity; but here I see lot more refactoring changes; even in non-related files (where the actual change is not expected); it will be nice to keep those changes minimal and focused on the issue being fixed...just my thoughts and I am fine with other way too...

The fix and the cleanup on on two separate commits to make reviewing easier. The other cleaned files are files affected by the correction of issues in a dependent class.

**Review by dschneider-pivotal — APPROVED** [review]

**Review by kirklund — APPROVED** [review]

**Review by gesterzhou — APPROVED** [review]

**Review by boglesby — APPROVED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java
Comment (nabarunnag) [inline geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java:1448]: Here we have completely removed the check for msg.clientVersion for null. Looking at the code there are some places where this valus is set to nul like in CommitCommand. Now the member version source is always sent, is this behavior change correct?

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java
Comment (nabarunnag) [inline geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java:1462]: Now sendVersionTags is always set to to true irrespective of if msg.clientVersion is null or not.

**Review by nabarunnag — CHANGES_REQUESTED** [review]
Few concerns regarding the changes in the behavior of toData in TXCommitMessage 
Changes may not be needed if a valid explanation is available.

**Comment by nabarunnag** [comment]
Also, I would request refactorings to be kept in separate PRs. 
- The caption JIRA mentions that this GEODE-8870 ticket is for removal of obsolete GFE_70 code but has GEODE-6588 along with it which has other major refactorings.
- Keeping the CL shorter will have faster approval / thorough review as the reviewer list smaller
- We view the final set of files changed as it is more convenient than looking at each and every commit individually, as the same file may have been modified in multiple commits.

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java:1448]: It would send the `VersionSource` if it didn't know the `clientVersion` or it was newer than v7.0. In this case the latter part is always true so the check for unknown or `null` `clientVersion` is redundant.

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java
Comment (pivotal-jbarrett) [reply geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java:1462]: For the same reason as above.

**Review by nabarunnag — APPROVED** [review]

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java
Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java:1448]: Thank you for the explanation

**Comment by pivotal-jbarrett** [comment]
> Also, I would request refactorings to be kept in separate PRs.
> 
> * The caption JIRA mentions that this GEODE-8870 ticket is for removal of obsolete GFE_70 code but has GEODE-6588 along with it which has other major refactorings.
> * Keeping the CL shorter will have faster approval / thorough review as the reviewer list smaller
> * We view the final set of files changed as it is more convenient than looking at each and every commit individually, as the same file may have been modified in multiple commits.

I think you should take this preference to a discussion on dev@geode. We have had similar discussions without an clear consensus. The closest we got was around not forcing squashes on PRs that include explicitly distinguished fix and cleanup commits, like this one.
https://markmail.org/message/hflkinvcxjg6stbo

**Comment by nabarunnag** [comment]
Sure but GEODE-6588 could have been another PRs and limit this PR to GEODE-8870

**Comment by pivotal-jbarrett** [comment]
> Sure but GEODE-6588 could have been another PRs and limit this PR to GEODE-8870

Is the issue that there is cleanup of files touched by the fix in this PR as separate commits or that those commits have a different JIRA issue associated with them?

**Pull request merged**
